### PR TITLE
Adding a method to removing timed gags from a player

### PR DESCRIPTION
**Pull request by overlordbleck**
Kinda yucky imo, would love some suggestions on how to clean this up, if at all.

### CODE REVIEW

File: lua/ulx/modules/sh/timed_gag.lua [hunk 1]
```diff
@@ -124,6 +124,14 @@ local function removeExpiredGagFromDatabase( ply )
     return SQLAction( "Deleting expired time gag", query, ply )
 end
 
+local function removeGagFromDatabase( ply )
```
Comment (PLally) [inline lua/ulx/modules/sh/timed_gag.lua]: removeExpiredGag could be replaced by this function unless logging `Deleting expired time gag` vs `Manually deleting time gag` is important
they run the same  query so maybe just a single function with the message `Deleting  timed gag` would be better

File: lua/ulx/modules/sh/timed_gag.lua [hunk 2]
```diff
@@ -298,8 +309,8 @@ local function updateGags()
 
     for ply, expiration in pairs( GaggedPlayers ) do
         if isValidPlayer( ply ) then
-            if gagIsExpired( expiration ) or not playerIsUlxGagged( ply ) then
-                removeExpiredGagFromDatabase( ply )
+            if gagIsExpired( expiration ) then
+                removeGagFromDatabase( ply )
                 ungagPlayer( ply )
             end
         else
```
Comment (justinmuskopf) [inline lua/ulx/modules/sh/timed_gag.lua]: L223 has an error - It still tries to call `removeExpiredGagFromDatabase`

File: lua/ulx/modules/sh/timed_gag.lua [hunk 3]
```diff
@@ -243,23 +243,34 @@ end
 
 -- ULX COMMAND SETUP --
```
Comment (justinmuskopf) [inline lua/ulx/modules/sh/timed_gag.lua]:
```suggestion
local function removeTimeGag(ply)
    removeGagFromDatabase( ply )
    ungagPlayer( ply )
end
```

File: lua/ulx/modules/sh/timed_gag.lua [hunk 4]
```diff
@@ -243,23 +243,34 @@ end
...
 
-local function timeGag( callingPlayer, targetPlayers, minutesToGag, reason )
-    ulx.fancyLogAdmin( callingPlayer, "#A gagged #T for #i minutes!", targetPlayers, minutesToGag )
+local function timeGag( callingPlayer, targetPlayers, minutesToGag, reason, shouldUngag )
+    if shouldUngag then
+        for _, ply in pairs( targetPlayers ) do
+            removeGagFromDatabase( ply )
+            ungagPlayer( ply )
+        end
```
Comment (justinmuskopf) [inline lua/ulx/modules/sh/timed_gag.lua]:
```suggestion
        for _, ply in pairs( targetPlayers ) do removeTimeGag( ply ) end
```

File: lua/ulx/modules/sh/timed_gag.lua [hunk 5]
```diff
@@ -298,8 +309,8 @@ local function updateGags()
 
     for ply, expiration in pairs( GaggedPlayers ) do
         if isValidPlayer( ply ) then
-            if gagIsExpired( expiration ) or not playerIsUlxGagged( ply ) then
-                removeExpiredGagFromDatabase( ply )
+            if gagIsExpired( expiration ) then
+                removeGagFromDatabase( ply )
                 ungagPlayer( ply )
             end
```
Comment (justinmuskopf) [inline lua/ulx/modules/sh/timed_gag.lua]:
```suggestion
            if gagIsExpired( expiration ) then removeTimeGag( ply ) end
```

**Review by justinmuskopf — DISMISSED** [review]

File: lua/ulx/modules/sh/timed_gag.lua (hunk 3 above)
Comment (loganlearner) [reply lua/ulx/modules/sh/timed_gag.lua]: Are you suggesting this be its own function rather than an opposite function?

**Review by brandonsturgeon — APPROVED** [review]

**Pull request merged**